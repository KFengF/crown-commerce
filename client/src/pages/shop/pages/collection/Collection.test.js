import React from 'react';
import { shallow } from 'enzyme';
import { CollectionPage } from './Collection.page';

describe('Collection page', () => {
    let mockTitle;
    let mockItems;
    let wrapper;

    beforeAll(() => {
        mockTitle = 'hats';
        mockItems = [
            { name: 'a', id: 1 },
            { name: 'b', id: 2 },
            { name: 'c', id: 3 },
            { name: 'd', id: 4 },
            { name: 'e', id: 5 },
        ];
        const mockProps = {
            collection: {
                title: mockTitle,
                items: mockItems
            }
        }
        wrapper = shallow(<CollectionPage { ...mockProps } />);
    });

    it('should render Collection page', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('collection prop', () => {
        it('should be the text hats in CollectionTitleH2', () => {
            expect(wrapper.find('CollectionTitleH2').text()).toBe(mockTitle);
        });

        it('should be five CollectionItem components', () => {
            expect(wrapper.find('Connect(CollectionItem)')).toHaveLength(5);
        });
    });
});