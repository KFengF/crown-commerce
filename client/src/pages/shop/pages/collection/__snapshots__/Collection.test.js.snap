// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection page should render Collection page 1`] = `ShallowWrapper {}`;
